src/stream.rs

```diff
@@ -116,6 +116,7 @@ impl VsockStream {
                 self.io.clear_write_ready(cx)?;
                 Poll::Pending
             }
+            Err(ref e) if e.kind() == ErrorKind::Interrupted => self.poll_write_priv(cx, buf),
```

Comment:
```suggestion
        loop {
            match self.io.get_ref().write(buf) {
                Err(e) if e.kind() == ErrorKind::WouldBlock => {
                    self.io.clear_write_ready(cx)?;
                    return Poll::Pending;
                }
                Err(e) if e.kind() == ErrorKind::Interrupted => continue,
                x => return Poll::Ready(x),
            }
        }
```

Reply:
Oh, you preferred the loop? I thought it was a bit more dangerous than just calling the same function.

Reply:
Not only about loop, it's about what to retry, I think we need retry the `self.io.get_ref().write` not `poll_write_priv `.